runtime/lua/vim/lsp/codelens.lua

```diff
@@ -181,6 +183,13 @@ function M.save(lenses, bufnr, client_id)
         api.nvim_buf_clear_namespace(b, ns, first_lnum, last_lnum)
       end,
     })
+    api.nvim_create_autocmd('LspDetach', {
```

Comment:
I think this will result in another autocmd definition on each Detach - Attach - Detach cycle. 

This could probably be defined only once on import globally and it can use the `ev.buf` property for the buffer number.

Reply:
Thanks for comment. Fixed as suggested.